pkg/blockstorage/azure/azuredisk.go

```diff
@@ -376,95 +357,98 @@ func (s *AdStorage) SnapshotGet(ctx context.Context, id string) (*blockstorage.S
 		return nil, errors.Wrapf(err, "SnapshotsClient.Get: Failed to get snapshot with ID %s", id)
 	}
 
-	return s.snapshotParse(ctx, snap), nil
+	return s.snapshotParse(ctx, snapRes.Snapshot), nil
 }
 
 func (s *AdStorage) VolumeParse(ctx context.Context, volume interface{}) (*blockstorage.Volume, error) {
-	vol, ok := volume.(azcompute.Disk)
+	vol, ok := volume.(armcompute.Disk)
 	if !ok {
-		return nil, errors.New(fmt.Sprintf("Volume is not of type *azcompute.Disk, volume: %v", volume))
+		return nil, errors.New(fmt.Sprintf("Volume is not of type *armcompute.Disk, volume: %v", volume))
 	}
 	encrypted := false
-	if vol.DiskProperties.EncryptionSettingsCollection != nil &&
-		vol.DiskProperties.EncryptionSettingsCollection.Enabled != nil {
-		encrypted = *vol.DiskProperties.EncryptionSettingsCollection.Enabled
+	if vol.Properties.EncryptionSettingsCollection != nil &&
+		vol.Properties.EncryptionSettingsCollection.Enabled != nil {
+		encrypted = *vol.Properties.EncryptionSettingsCollection.Enabled
 	}
 	tags := map[string]string{"": ""}
 	if vol.Tags != nil {
-		tags = azto.StringMap(vol.Tags)
+		tags = blockstorage.StringMap(vol.Tags)
 	}
-	az := azto.String(vol.Location)
-	if z := azto.StringSlice(vol.Zones); len(z) > 0 {
-		az = az + "-" + z[0]
+	az := blockstorage.StringFromPtr(vol.Location)
+	if z := vol.Zones; len(z) > 0 {
+		az = az + "-" + *(z[0])
 	}
 
 	return &blockstorage.Volume{
 		Type:         s.Type(),
-		ID:           azto.String(vol.ID),
+		ID:           blockstorage.StringFromPtr(vol.ID),
 		Encrypted:    encrypted,
-		SizeInBytes:  azto.Int64(vol.DiskSizeBytes),
+		SizeInBytes:  blockstorage.Int64(vol.Properties.DiskSizeBytes),
 		Az:           az,
 		Tags:         blockstorage.MapToKeyValue(tags),
-		VolumeType:   string(vol.Sku.Name),
-		CreationTime: blockstorage.TimeStamp(vol.DiskProperties.TimeCreated.ToTime()),
-		Attributes:   map[string]string{"Users": azto.String(vol.ManagedBy)},
+		VolumeType:   string(*vol.SKU.Name),
+		CreationTime: blockstorage.TimeStamp(*vol.Properties.TimeCreated),
+		Attributes:   map[string]string{"Users": blockstorage.StringFromPtr(vol.ManagedBy)},
 	}, nil
 }
 
 func (s *AdStorage) SnapshotParse(ctx context.Context, snapshot interface{}) (*blockstorage.Snapshot, error) {
-	if snap, ok := snapshot.(azcompute.Snapshot); ok {
+	if snap, ok := snapshot.(armcompute.Snapshot); ok {
 		return s.snapshotParse(ctx, snap), nil
 	}
-	return nil, errors.New(fmt.Sprintf("Snapshot is not of type *azcompute.Snapshot, snapshot: %v", snapshot))
+	return nil, errors.New(fmt.Sprintf("Snapshot is not of type *armcompute.Snapshot, snapshot: %v", snapshot))
 }
 
-func (s *AdStorage) snapshotParse(ctx context.Context, snap azcompute.Snapshot) *blockstorage.Snapshot {
+func (s *AdStorage) snapshotParse(ctx context.Context, snap armcompute.Snapshot) *blockstorage.Snapshot {
 	vol := &blockstorage.Volume{
 		Type: s.Type(),
-		ID:   azto.String(snap.SnapshotProperties.CreationData.SourceResourceID),
+		ID:   *snap.ID,
```

Comment:
[nil pointer] ditto on guard for `nil` pointer

Reply:
Done